apps/sms/test/marionette/mocks/mock_navigator_moz_mobile_message.js

```diff
@@ -161,11 +183,29 @@ Services.obs.addObserver(function(document) {
         onerror: null
       }, window));
 
-      window.setTimeout(function() {
+      getStorage().then(function(storage) {
         if (typeof cursor.onsuccess === 'function') {
-          cursor.onsuccess.call(cursor);
+          var threads = storage.threads.values();
+
+          cursor.continue = function() {
+            var iteratorResult = threads.next();
+
+            if (!iteratorResult.done) {
+              cursor.result = iteratorResult.value;
+            } else {
+              cursor.done = true;
+              cursor.continue = null;
+              cursor.result = null;
+            }
+
+            window.setTimeout(function nextTick() {
```

Comment:
So as promised :) Here I tried to use ```Promise.resolve().then(cursor.onsuccess.call(cursor))``` instead, but on ~50 iterations it fails with "too much recursion" exception.

So we have the following flow here:
```
--- continue() (first time)
--- --- Promise.resolve().then(() => onsuccess())
--- --- --- continue()
--- --- --- --- Promise.resolve().then(() => onsuccess())
--- --- --- --- --- continue()
...........
```

Looks like promise somehow keeps stack of everything inside callback even if ```Promise.resolve``` not returned from ```then``` callback.